### PR TITLE
docs: add remote development document

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Add remote development document
TODO: update all images after extension rebranding and wording review

### CODE REVIEW

File: remote-finetuning.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+You can monitor the progress of the provision via the link that is displayed in the output channel.
+![Provision Progress](Images/remote/provision-progress.png)
+
+## Run
+To start the remote fine-tuning job, execute the `AI Toolkit: Run fine-tuning` command.
+![Run Fine-tuning](Images/remote/run-finetuning.png)
+
+Upon running this command, the extension will do the following operations:
+1. Synchronize your workspace with Azure Files.
+1. Start the ACA job using the commands specified in `./infra/fintuning.config.json`.
```
Comment (swatDong) [inline remote-finetuning.md:24]: Just call it Azure Container App, no "ACA"

File: remote-finetuning.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+1. Display the job streaming log if the job has started running. 
+    ![Run Log](Images/remote/run-log.png)
+    > **Note:** The job might be queued if there are insufficient resources available. If the log fails to display when the job starts, you can wait for a while and then execute the `AI Toolkit: Show the running fine-tuning job streaming logs` command to re-connect to the streaming log.
+    
+During this process, QLoRA will be used for fine-tuning, and will create LoRA adapters for the model to use during inference.
+The results of the fine-tuning will be stored in the Azure Files.
```
Comment (swatDong) [inline remote-finetuning.md:30]: Add section for accessing those output files

File: remote-finetuning.md (hunk 1 above)
Comment (swatDong) [reply remote-finetuning.md:24]: Fine to use "aca" in variable naming, but not in description

File: remote-inference.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+![Fine-tune complete](Images/remote/finetune-res.png)
+
+## Provision Azure Resources
+Similar to the fine-tuning process, you need to set up the Azure Resources for remote inference by executing the `AI Toolkit: Provision Azure Container Apps for inference.` from the command palette. During this setup, you will be asked to select your Azure Subscription and resource group.  
+![Provision Inference Resource](Images/remote/provision-inference.png)
+   
+By default, the subscription and the resource group for inference should match those used for fine-tuning. The inference will use the same Azure Container App Environment and access the model and model adapter stored in Azure Files, which were generated during the fine-tuning step. 
+
+Once provisioning is successfully completed, you can find the web API endpoint under `ACA_APP_ENDPOINT` within `./infra/inference.config.json`. You are now ready to evaluate the model using this endpoint.
```
Comment (kuojianlu) [inline remote-inference.md:12]: No need to run deploy command?

File: remote-finetuning.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
+# Fine-Tuning models remotely
+Model fine-tuning in machine learning involves subtly adjusting a pre-existing model, which was initially trained on a larger dataset, to perform a similar but new task using a smaller dataset. If you lack local computing resources, such as a GPU, you can perform this fine-tuning remotely using AI Toolkit and Azure Container App.
```
Comment (supkasar) [inline remote-finetuning.md:2]: Proposed:
Model fine tuning in machine learning involves subtly adjusting an existing model, originally trained on a larger dataset, to perform a similar but new task using a smaller dataset. If you don't have local computing resources like a GPU, you can do this fine tuning remotely with AI Toolkit and Azure Container App.

File: remote-finetuning.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+## Prerequisites
+1. To run the model fine-tuning in your remote Azure Container App Environment, you need to make sure your subscription have enough GPU capacity amount. Submit a [support ticket](https://azure.microsoft.com/support/create-ticket/) to request the capacity amount required for your application. [Learn More about GPU capacity](https://learn.microsoft.com/en-us/azure/container-apps/workload-profiles-overview)
```
Comment (supkasar) [inline remote-finetuning.md:5]: Proposed:
To run the model fine tuning in your Azure Container App Environment, make sure your subscription has enough GPU capacity. Submit a [support ticket] (https://azure.microsoft.com/support/create-ticket/) to request the required capacity for your application. [Get More Info about GPU capacity](https://learn.microsoft.com/en-us/azure/container-apps/workload-profiles-overview)

File: remote-finetuning.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+2. Make sure you have a [HuggingFace account](https://huggingface.co/) and [generate an access token](https://huggingface.co/docs/hub/security-tokens) if you are using private dataset on HuggingFace or your base model needs access control.
+3. Accept the LICENSE on HuggingFace if you are fine-tuning Mistral or Llama. 
+4. If you enable [Weights & Biases](https://wandb.ai/site), make sure you have a Weights & Biases account and [retrieve your API Key](https://wandb.ai/authorize)
+
+## Provision Azure Resources
+To get started, you need provision the Azure Resource for remote fine-tuning. This can be done by running the `AI Toolkit: Provision Azure Container Apps job for fine-tuning.` from command palette. During this process, you will be prompted to select your Azure Subscription and resource group.
```
Comment (supkasar) [inline remote-finetuning.md:11]: Proposed:
To get started, you need to provision the Azure Resource for remote fine-tuning. Do this by running the AI Toolkit: Provision Azure Container Apps job for fine-tuning. from the command palette.

File: remote-finetuning.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    > **Note:** The job might be queued if there are insufficient resources available. If the log fails to display when the job starts, you can wait for a while and then execute the `AI Toolkit: Show the running fine-tuning job streaming logs` command to re-connect to the streaming log.
+    
+During this process, QLoRA will be used for fine-tuning, and will create LoRA adapters for the model to use during inference.
+The results of the fine-tuning will be stored in the Azure Files.
+
+## View and Query Logs on Azure
+
+After fine-tuning job was triggered, you can also view logs on Azure by clicking the "*Open Logs in Azure Portal*" button from the VSCode notification.
+
+Or, if you already opened the Azure Portal, you can find job history from the "*Execution history*" panel to the Azure Container Apps job.
```
Comment (supkasar) [inline remote-finetuning.md:36]: Proposed:
Or, if you've already opened the Azure Portal, find job history from the "*Execution history*" panel to the Azure Container Apps job.

File: remote-finetuning.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+![Job Execution History](Images/remote/finetune-job-history.png)
+
+There are two types of logs, "*Console*" and "*System*".
+- Console logs, which are emitted by your app, including `stderr` and `stdout` messages. That's also what you may already see in previous section's streaming logs.
```
Comment (supkasar) [inline remote-finetuning.md:41]: Proposed: 
Console logs are messages from your app, including stderr and stdout messages. This is what you might have already seen in the streaming logs section.

File: remote-finetuning.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+- System logs, which are emitted by the Container Apps service, including the status of service level events.
```
Comment (supkasar) [inline remote-finetuning.md:42]: Proposed:
System logs are messages from the Container Apps service, including the status of service-level events.

File: remote-finetuning.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+| Folder | Contents |
+| ------ |--------- |
+| `infra` | Contains all necessary configurations for remote operations. |
+| `infra/provision/finetuning.parameters.json` | Holds parameters for the bicep templates, used for provisioning Azure resources for fine-tuning. |
+| `infra/provision/finetuning.bicep` | Contains templates for provisioning Azure resources for fine-tuning. |
+| `infra/finetuning.config.json` |The configuration file, generated by the `AI Toolkit: Provision Azure Container Apps job for fine-tuning.` command. It is used as input for other remote command palettes. |
+
+## Advanced usage
+### Configuring Secrets for fine-tuning in Azure Container Apps
+Azure Container App Secrets offer a secure way to store and manage sensitive data within Azure Container Apps. This feature fulfills the need for a secure environment to handle sensitive data such as HuggingFace tokens and Weights & Biases API keys. If you need to set these values, AI Toolkit provides a command palette to input the secrets into the provisioned Azure container app job (as stored in `./finetuning.config.json`). These secrets are then set as **environment variables** in all the containers.
```
Comment (supkasar) [inline remote-finetuning.md:62]: Proposed:
Azure Container App Secrets provide a secure way to store and manage sensitive data within Azure Container Apps, like HuggingFace tokens and Weights & Biases API keys. Using AI toolkit's command palette, you can input the secrets into the provisioned Azure container app job (as stored in `./finetuning.config.json`). These secrets are then set as **environment variables** in all containers.

File: remote-finetuning.md (hunk 1 above)
Comment (XiaofuHuang) [reply remote-finetuning.md]: Replaced all `ACA` to be `Azure Container Apps`

File: remote-finetuning.md (hunk 2 above)
Comment (XiaofuHuang) [reply remote-finetuning.md:27]: Added this section.

File: remote-inference.md (hunk 3 above)
Comment (XiaofuHuang) [reply remote-inference.md]: Renamed `Update code for inference` to be `Deploy for inference`

File: remote-finetuning.md (hunk 4 above)
Comment (XiaofuHuang) [reply remote-finetuning.md:2]: Updated.

File: remote-finetuning.md (hunk 5 above)
Comment (XiaofuHuang) [reply remote-finetuning.md]: Updated.

File: remote-finetuning.md (hunk 6 above)
Comment (XiaofuHuang) [reply remote-finetuning.md]: Updated.

File: remote-finetuning.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+2. Make sure you have a [HuggingFace account](https://huggingface.co/) and [generate an access token](https://huggingface.co/docs/hub/security-tokens) if you are using private dataset on HuggingFace or your base model needs access control.
+3. Accept the LICENSE on HuggingFace if you are fine-tuning Mistral or Llama. 
+4. If you enable [Weights & Biases](https://wandb.ai/site), make sure you have a Weights & Biases account and [retrieve your API Key](https://wandb.ai/authorize)
+
+## Provision Azure Resources
+To get started, you need provision the Azure Resource for remote fine-tuning. This can be done by running the `AI Toolkit: Provision Azure Container Apps job for fine-tuning.` from command palette. During this process, you will be prompted to select your Azure Subscription and resource group.
+
+![Provision Fine-Tuning](Images/remote/provision-finetune.png)
+
+You can monitor the progress of the provision via the link that is displayed in the output channel.
```
[the remark this answers is not in the document]
Comment (XiaofuHuang) [reply remote-finetuning.md]: Updated.

File: remote-finetuning.md (hunk 7 above)
Comment (XiaofuHuang) [reply remote-finetuning.md]: Updated

File: remote-finetuning.md (hunk 8 above)
Comment (XiaofuHuang) [reply remote-finetuning.md:48]: Updated

File: remote-finetuning.md (hunk 9 above)
Comment (XiaofuHuang) [reply remote-finetuning.md:49]: Updated.

File: remote-finetuning.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+Or, if you already opened the Azure Portal, you can find job history from the "*Execution history*" panel to the Azure Container Apps job.
+
+![Job Execution History](Images/remote/finetune-job-history.png)
+
+There are two types of logs, "*Console*" and "*System*".
+- Console logs, which are emitted by your app, including `stderr` and `stdout` messages. That's also what you may already see in previous section's streaming logs.
+- System logs, which are emitted by the Container Apps service, including the status of service level events.
+
+To view and query you logs, clicking the "*Console*" button and you will be navigated to the Log Analytics page. There you can view all the logs and write your own queries.
```
[the remark this answers is not in the document]
Comment (XiaofuHuang) [reply remote-finetuning.md:51]: Updated

File: remote-finetuning.md (hunk 10 above)
Comment (XiaofuHuang) [reply remote-finetuning.md:83]: Updated

**Review by swatDong — APPROVED** [review]

**Pull request merged**